### PR TITLE
ci: added a new high priority queue to mergify

### PR DESCRIPTION
**Pull request by frazarshad**
refs: #8396

## Description

This PR adds a new high-priority queue to mergify. A high priority queue allows us to merge urgent PRs before low priority ones, essentially speeding up patch/hotfix release.
This does not bypass any CI checks for merging so all the normal standard is still followed even for high priority queues

Resources: https://docs.mergify.com/merge-queue/multi/

The way to use this new high priority queue is to add a new label `priority:high` github and add that on high priority PRs. 

### Fallback:
In case the new queue does not work as expected, the original queue is sill present and working

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -1,5 +1,23 @@
 # Linear queue for the main branch.
 queue_rules:
+  - name: high_priority
+    commit_message_template: |-
+      {{ title }} (#{{ number }})
+
+      {{ body | trim }}
+    queue_conditions:
```
Comment (mhofman) [inline .mergify.yml:8]: I think we should maintain the `check-success=breakage` condition

File: .mergify.yml [hunk 2]
```diff
@@ -1,5 +1,23 @@
 # Linear queue for the main branch.
 queue_rules:
+  - name: high_priority
```
Comment (mhofman) [inline .mergify.yml]: Let's name this `high_priority_main`

**Review by mhofman — COMMENTED** [review]
This is great, some small changes requested. Also is there a PR on the experiment repo where this is tested? (please keep that repo in sync with agoric-sdk as well)

While you're in this neighborhood, I'm wondering if we should address the multiple queue part of https://github.com/Agoric/agoric-sdk/issues/8578 now that have better separation between queue and merge rules.

In particular I think we just need a couple more queues, where the `base` condition is not `master`. I think we need one for all `dev-*` branches, and one for all `release-*` branches.

Bonus points for having these queues independent of the main queue, but I'm also ok with a single queue and some precedence like high_priority_main > release > main > dev

**Comment by frazarshad** [comment]
>  Also is there a PR on the experiment repo where this is tested? (please keep that repo in sync with agoric-sdk as well)

This is tested on my personal mergify repo. but i've created a PR on the agoric mergify experiments repo as you suggested:
created a PR for this: https://github.com/Agoric/mergify-experiements/pull/18

> While you're in this neighborhood, I'm wondering if we should address the multiple queue part of #8578 now that have better separation between queue and merge rules.

Working on it in a seperate PR

**Comment by frazarshad** [comment]
This PR has been tested on https://github.com/Agoric/mergify-experiements/pull/18. The new mergify config was merged after which two PRs on[ low priority](https://github.com/Agoric/mergify-experiements/pull/19) and one [high priority](https://github.com/Agoric/mergify-experiements/pull/20) was enqueued one after the other. The high priority one was forcibly enqued to the top and was checked before the low priority one.
https://dashboard.mergify.com/queues?branch=main&repository=mergify-experiements&login=Agoric
image

**Review by mhofman — APPROVED** [review]
Thanks for creating examples of the new high priority logic in the experiments repo!

**Pull request merged**